### DISCUSSION TITLE
Use fetch to request the next.js self-built api file in the app route, showing that the request failed.

### DISCUSSION THREAD

**Discussion by x751685875** [vercel/next.js#58077] [App Router] [answered]
Use fetch to request the next.js self-built api file in the app route, showing that the request failed.

I used app.localhost:3000 for development locally, and an error was reported when I requested it.

My request url is http://app.localhost:3000/api/find-profile

```
 ○ Compiling /[domain]/page ...
 ✓ Compiled /[domain]/page in 1670ms (2449 modules)
 ⚠ Fast Refresh had to perform a full reload due to a runtime error.
 ⨯ Internal error: TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11372:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Cause: Error: getaddrinfo ENOTFOUND app.localhost
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26)
    at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'app.localhost'
}
 ⨯ Internal error: TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11372:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
digest: "3747308182"
``

**Comment by icyJoseph** [comment, marked as answer]
Maybe read through here: 

- https://nextjs-faq.com/fetch-api-in-rsc

In general, if you are in a server component, or server side in general, you should just fetch the data you need, from its source, no need to make extra jumps. 

Note that, Nuxt js, a Vue based framework, does allow for such a model, by invoking the functions using their special `$fetch` function: https://nuxt.com/docs/guide/concepts/server-engine#direct-api-calls

**Reply by makeable** [reply]
All the reasons "against" also apply for any external API. You could implement an API as a separate instance, and then it's considered OK, so why not be OK as part of the same instance? The server side fetch could bypass the network entirely and handle this for the user when they call a relative route. As is, It's an over-opinionated view of how my data layer should be constructed, and doesn't nothing other than increase complexity IMHO.
